cpp/src/utilities/traits.cpp

```diff
@@ -0,0 +1,399 @@
+/*
+ * Copyright (c) 2019-2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuda_runtime.h>
```

Comment:
What is being used from the CUDA runtime here? I didn't see it.

Reply:
This is needed for CUDA declarations (e.g. `__device__`) that appear in header files like `string_view.hpp` and `type_dispatcher.hpp` when included in a `.cpp` file.

Reply:
Ah, of course. Thanks.